npp.sh

```diff
@@ -0,0 +1,127 @@
+    gray|grey) FontSmoothing=2; FontSmoothingOrientation=1; FontSmoothingType=1;;
+    bgr)       FontSmoothing=2; FontSmoothingOrientation=0; FontSmoothingType=2;;
+    rgb)       FontSmoothing=2; FontSmoothingOrientation=1; FontSmoothingType=2;;
+    *)         return;;
+  esac
+  echo "Setting font smoothing to $1"
+  REG_FILE=$(mktemp)
+  cat > "$REG_FILE" <<EOL
+REGEDIT4
+[HKEY_CURRENT_USER\\Control Panel\\Desktop]
+"FontSmoothing"="${FontSmoothing}"
+"FontSmoothingGamma"=dword:00000578
+"FontSmoothingOrientation"=dword:0000000${FontSmoothingOrientation}
+"FontSmoothingType"=dword:0000000${FontSmoothingType}
+EOL
+  wine64 regedit /s "$REG_FILE"
+  rm "$REG_FILE"
+}
+
+function select_ui_font() {
+  find /run/host/fonts /run/host/local-fonts -iname '*sans*' -exec fc-scan -f "%{family}\t:%{style}:%{lang}\n" {} \; |\
+    grep -E '\s+:Regular:[a-z\|\-]*en[a-z\|\-]*$' | grep -Ev 'Mono|CJK' | awk 'BEGIN {FS="\t"}; {print $1}' | sort -u | {
+      LIST_ITEMS=""
+      unset FONTS
+      readarray FONTS
+    for CURRENT_FONT in "${FONTS[@]}"; do
+      CURRENT_FONT=$(echo "$CURRENT_FONT" | tr -d '\n')
+      LIST_ITEMS="${LIST_ITEMS} FALSE '${CURRENT_FONT}'"
+    done
+    sh -c "zenity --width=600 --height=400 --list --text='UI Font' --radiolist --column='' --column='Font Family' TRUE Default ${LIST_ITEMS}"
+    }
+}
+
+function set_ui_font() {
+  echo "Setting UI font to ${1}"
+  wine64 REG ADD "HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\FontSubstitutes" /v "MS Shell Dlg" /t REG_SZ /d "$1" /f
+  wine64 REG ADD "HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\FontSubstitutes" /v "MS Shell Dlg 2" /t REG_SZ /d "$1" /f
+}
+
+function initialize_wine() {
+  # initialize prefix
+  wineboot
+
+  # prevent wine from making new file associations
+  wine64 reg add "HKLM\Software\Microsoft\Windows\CurrentVersion\RunServices" /v winemenubuilder /t REG_SZ /d "C:\windows\system32\winemenubuilder.exe -r" /f
+
+  install_color_theme
+
+  SELECTED_DPI=$(zenity --width=600 --height=400 --list --text="Screen Resolution" --radiolist --column="" --column="DPI" TRUE 96 FALSE 120 FALSE 144 FALSE 168 FALSE 192 FALSE 216 FALSE 240 FALSE 288 FALSE 336 FALSE 384 FALSE 432 FALSE 480)
+  if [ "$SELECTED_DPI" ]; then
+    set_dpi $SELECTED_DPI
+  fi
+
+  SELECTED_FONT_SMOOTHING=$(zenity --width=600 --height=400 --list --text="Font Smoothing" --radiolist --column="" --column="Sub-pixel Rendering" TRUE default FALSE disable FALSE gray FALSE bgr FALSE rgb)
+  if [ "$SELECTED_FONT_SMOOTHING" ]; then
+    set_font_smoothing $SELECTED_FONT_SMOOTHING
+  fi
+
+  SELECTED_UI_FONT=$(select_ui_font)
+  if [ ! -z "$SELECTED_UI_FONT" ] && [ "$SELECTED_UI_FONT" != "Default" ]; then
+    set_ui_font "$SELECTED_UI_FONT"
+  fi
+}
+
+# initialize wine and install npp if prefix directory doesn't exist or is empty
+if [ ! -d "$WINEPREFIX/drive_c" ]; then
+  initialize_wine
+  mkdir -p "$NPP_ROOT"
+  tar xf /app/data/npp.tar.xz -C "$NPP_ROOT"
```

Comment:
This feels very wrong. Will every wine app gonna ship like this? Who gonna manage bloat pilling in the user's home? How the app manages updates?  
You can inject a version file to compare against, I've done this [here](https://github.com/tinywrkb/flatpaks/blob/master/com.ti.ccstudio/com.ti.ccstudio.yaml#L55), see also [init script](https://github.com/tinywrkb/flatpaks/blob/master/com.ti.ccstudio/resources/ccstudio#L58), but not in a Flathub distributed package, and it still feels very wrong.  
Wine should have proper Flatpak support instead of making the user's home a mismanaged installation target, especially when the user does not create the Wine prefix, unlike when using Lutris.